android/app/src/main/java/com/microsoft/codepush/common/core/CodePushBaseCore.java

```diff
@@ -0,0 +1,1043 @@
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.AWAITING_USER_ACTION;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.CHECKING_FOR_UPDATE;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.DOWNLOADING_PACKAGE;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.SYNC_IN_PROGRESS;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.UNKNOWN_ERROR;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.UPDATE_IGNORED;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.UPDATE_INSTALLED;
+import static com.microsoft.codepush.common.enums.CodePushSyncStatus.UP_TO_DATE;
+import static com.microsoft.codepush.common.enums.CodePushUpdateState.LATEST;
+import static com.microsoft.codepush.common.enums.CodePushUpdateState.PENDING;
+import static com.microsoft.codepush.common.enums.CodePushUpdateState.RUNNING;
+
+/**
+ * Base core for CodePush. Singleton.
+ */
+public abstract class CodePushBaseCore {
+
+    /**
+     * Deployment key for checking for updates.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected String mDeploymentKey;
+
+    /**
+     * CodePush server URL.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected static String mServerUrl = "https://codepush.azurewebsites.net/";
+
+    /**
+     * Public key for code signing verification.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected static String mPublicKey;
+
+    /**
+     * Entry point for application.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected final String mAppEntryPoint;
+
+    /**
+     * Application context.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected final Context mContext;
+
+    /**
+     * Indicates whether application is running in debug mode.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected final boolean mIsDebugMode;
+
+    /**
+     * Current app version.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected String mAppVersion;
+
+    /**
+     * Current state of CodePush update.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected CodePushState mState;
+
+    /**
+     * Used utilities.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected CodePushUtilities mUtilities;
+
+    /**
+     * Used managers.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected CodePushManagers mManagers;
+
+    /**
+     * Used listeners.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected CodePushListeners mListeners;
+
+    /**
+     * Instance of {@link CodePushConfirmationDialog}.
+     */
+    @SuppressWarnings("WeakerAccess")
+    protected CodePushConfirmationDialog mConfirmationDialog;
+
+    /**
+     * Self-reference to the current instance.
+     */
+    @SuppressLint("StaticFieldLeak")
+    @SuppressWarnings("WeakerAccess")
+    protected static CodePushBaseCore mCurrentInstance;
```

Comment:
Why do we need currentInstance? Is it used somewhere?

Reply:
@annakocheshkova yeah, I will delete it, thx.